### ISSUE TITLE
getting az not found on pip install

### ISSUE THREAD

**Issue by jonbgallant** [jonbgallant/azure-iot-edge-dev-tool#44]

**Comment by jonbgallant** [comment]
More info:

```
C:\edge-projects>iotedgedev solution edgesolution1
 
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\iotedgedev.exe\__main__.py", line 5, in <module>
  File "c:\python27\lib\site-packages\iotedgedev\cli.py", line 12, in <module>
    from .azurecli import AzureCli
  File "c:\python27\lib\site-packages\iotedgedev\azurecli.py", line 10, in <module>
    from azure.cli.core import get_default_cli
ImportError: cannot import name get_default_cli

```

```
azure-cli (2.0.28)
.....
….
Extensions:
azure-cli-iot-ext (0.4.1)
Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\igorpag\.azure\cliextensions'
Python (Windows) 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 17:54:52) [MSC v.1900 32 bit (Intel)]
```

**Comment by jonbgallant** [comment]
@digimaun Should we add azure-cli-core to requirements.txt and setup.py? 

@paloukari

**Comment by jonbgallant** [comment]
pip install azure-cli-core did not solve this issue

I tried on a clean machine and **did not see this issue**.

Something interesting....

User was using Python 2.7.14, buy the az message above says 3.6.1

**Comment by jonbgallant** [comment]
The current workaround is to use the IoT Edge Dev Container:
docker run -it -v /var/run/docker.sock:/var/run/docker.sock -v c:/temp/iotedge:/iotedge jongallant/iotedgedev

**Comment by digimaun** [comment]
azure-cli-core is already a pulled in dependency from setup.py and requirements.txt assets, it does not need to be explicitly added.

With Az CLI msi install on Windows, the installer creates a separate python environment for Az assets, which is the reason you see python 3.6.1 from the az command, with a concurrent user installed python 2.7 environment.

**Comment by digimaun** [comment]
The cause of this issue is likely due to some path/pythonpath clobbering. I am curious what the users path/pythonpath looks like. The solution is to ensure the right python environment is executed against.

**Comment by digimaun** [comment]
Found out the post install hook is not reliably executing on py 27 environment so the az boot script (which updates python path) doesn't run in the target environment (and if you have another az installation set on your env path that would get run).

**Comment by jonbgallant** [comment]
Fixed in 0.73.0
